File: advanced/Mixed-Treaty/Data_Processing/data-processing.py
import os

def process_data(path):
    new_list= []
    if not os.path.exists(path):
        return "The path doesn't exist"
    else:
        with open(path, "r") as read_from_file:
            header = read_from_file.readline()
            # case empty file should result in empty file
            if header == "":
                return None
            # append the line in both case:
            if header.find("\n") != -1:
                new_list.append("Firstname,Lastname\n")
            else:
                new_list.append("Firstname,Lastname")
            # or just: new_list.append("Firstname,Lastname")
            
            #rest of the lines in the document
            lines = read_from_file.readlines()
            for line in lines:
                index_semicolon = line.find(";")

                # case lastname; firstname
                if index_semicolon != -1:
                    if line.find("\n") != -1:
                        firstname = line[index_semicolon + 2:-1]
                        lastname = line[:index_semicolon] + "\n"
                    else:
                        firstname = line[index_semicolon + 2:]
                        lastname = line[:index_semicolon]
                    new_list.append("{},{}".format(firstname, lastname))

                # case empty line
                elif line == "\n":
                    new_list.append(",\n")

                # case firstname lastname
                else:
                    index_space = line.find(" ")
                    firstname = line[:index_space]
                    lastname = line[index_space + 1:]
                    new_list.append("{},{}".format(firstname, lastname))
        
        # writing the collected data of new_list into the same text-file again
        with open(path, "w") as write_to_file:
            for line in new_list:
                write_to_file.write(line)

# write or copy-paste the path below
path = "/Users/.../test-js/advanced/Mixed-Treaty/Data_Processing/test.txt"
print(process_data(path))

File: advanced/Mixed-Treaty/Data_Processing/data-processing.js

// fs for file system to work on files in operatig-system 
const fs = require('fs');

function checkFileExists(filePath) {
  try {
    // Check if the file exists
    fs.accessSync(filePath, fs.constants.F_OK);
    return true; // File exists
  } catch (err) {
    return false; // File does not exist
  }
}

function processData(path) {
    let collectionString= ""
    if (!checkFileExists(path)){
        return "The path doesn't exist"
    }else{
        fs.readFile(path, (err, input) =>{ 
            if (err) throw (err);
            const inputString = input.toString()
            const inputArray= inputString.split("\n")
            collectionString+= "Firstname,Lastname\n"
            // checking the falls of ";", space " ", and extra falls in else
            for (let i=0; i< inputArray.length; i++){
                item = inputArray[i]
                if (item === ""){
                    collectionString+= ",\n"
                } else if (item.includes(';')){
                    let line = item.split("; ")
                    const firstName = line[1]
                    const lastName= line[0]
                    const fullName= firstName+","+lastName+"\n"
                    collectionString+= fullName
                } else if (item.includes(" ")){
                    let line = item.split(" ")
                    const firstName = line[0]
                    const lastName= line[1]
                    const fullName= firstName+","+lastName+"\n"
                    collectionString+= fullName
                } else {continue}
            }
            // deleting the last two characters which is \n: new line, in case it is needed
            collectionString = collectionString.slice(0, -2);

            // writing the processed data into the same file/path
            fs.writeFile(path, collectionString, "utf8", (err) => {
                if (err){ console.log("Error of kind: ", err); }
                else { console.log("Data written into file: ", collectionString)}
            });
        });
    }
}


// write or copy-paste the path below
const path = "/Users/ari/Documents/gitfolder/test-js/advanced/Mixed-Treaty/Data_Processing/test.txt"
processData(path);